google-beta/resource_compute_snapshot.go

```diff
@@ -1,220 +1,536 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file in
+//     .github/CONTRIBUTING.md.
+//
+// ----------------------------------------------------------------------------
+
 package google
 
 import (
 	"fmt"
 	"log"
+	"reflect"
+	"strconv"
 	"time"
 
+	"github.com/hashicorp/terraform/helper/customdiff"
 	"github.com/hashicorp/terraform/helper/schema"
-	"google.golang.org/api/compute/v1"
-	"google.golang.org/api/googleapi"
+	compute "google.golang.org/api/compute/v1"
 )
 
+func customDiffComputeSnapshotSnapshotEncryptionKeys(diff *schema.ResourceDiff, meta interface{}) error {
+	// oC -> Old Convenience Value
+	// nC -> New Convenience Value
+	// oN -> Old New Field Value
+	// nN -> New New Field Value
+	oC, nC := diff.GetChange("snapshot_encryption_key_raw")
+	oN, nN := diff.GetChange("snapshot_encryption_key.0.raw_key")
+
+	if nC != "" && nN != "" {
+		return fmt.Errorf("can't use snapshot_encryption_key_raw and snapshot_encryption_key.0.raw_key at the same time." +
+			"If you're removing snapshot_encryption_key.0.raw_key, set the value to \"\" instead. This is due to limitations in terraform plan.")
+	}
+
+	// Either field (convenience or new) has a value
+	// and then has another different value, so we ForceNew.
+	// We need to handle _EVERY_ ForceNew case in this diff
+	if oC != "" && nC != "" && oC != nC {
+		return diff.ForceNew("snapshot_encryption_key_raw")
+	}
+
+	if oN != "" && nN != "" && oN != nN {
+		return diff.ForceNew("snapshot_encryption_key.0.raw_key")
+	}
+
+	// Our resource isn't using either field, then uses one;
+	// ForceNew on whichever one is now using it.
+	if (oC == "" && oN == "" && nC != "") || (oC == "" && oN == "" && nN != "") {
```

Comment:
This could also be `oC == "" && oN == "" && (nC != "" || nN != "")`, which is just a little less repetition. Not worth changing if it makes more sense to you this way.

Reply:
I have a slight preference for the shape I chose - it's a little less nested and easier for me personally to read.

Reply:
👍